src/plugins/messageLinkEmbeds.ts

```diff
@@ -0,0 +1,344 @@
+            attachments.push({
+                height: e.thumbnail!.height,
+                width: e.thumbnail!.width,
+                url: e.url!
+            });
+        }
+    });
+    return attachments;
+}
+
+const noContent = (attachments: number, embeds: number): string => {
+    if (!attachments && !embeds) return "";
+    if (!attachments) return `[no content, ${embeds} embed${embeds !== 1 ? "s" : ""}]`;
+    if (!embeds) return `[no content, ${attachments} attachment${attachments !== 1 ? "s" : ""}]`;
+    return `[no content, ${attachments} attachment${attachments !== 1 ? "s" : ""} and ${embeds} embed${embeds !== 1 ? "s" : ""}]`;
+};
+
+const computeWidthAndHeight = (width: number, height: number) => {
+    const maxWidth = 400, maxHeight = 300;
+    let newWidth: number, newHeight: number;
+    if (width > height) {
+        newWidth = Math.min(width, maxWidth);
+        newHeight = Math.round(height / (width / newWidth));
+    } else {
+        newHeight = Math.min(height, maxHeight);
+        newWidth = Math.round(width / (height / newHeight));
+    }
+    return { width: newWidth, height: newHeight };
+};
+
+export default definePlugin({
+    name: "MessageLinkEmbeds",
+    description: "Embeds message links",
+    authors: [{
+        name: "ActuallyTheSun",
+        id: 406028027768733696n
+    }],
+    dependencies: ["MessageAccessoriesAPI"],
+    patches: [{
+        find: "_messageAttachmentToEmbedMedia",
+        replacement: {
+            match: /var .{1,2}=(.{1,2})\.channel,.{1,2}=.{1,2}\.message,.{1,2}=.{1,2}\.renderSuppressEmbeds/,
+            replace: (orig, props) =>
+                `${props}.message.embeds=Vencord.Plugins.plugins.MessageLinkEmbeds.generateRichEmbeds(${props}.message);${orig}`
+        }
+    },
+    {
+        find: "().embedCard",
+        replacement: [{
+            match: /{"use strict";(.{0,10})\(\)=>(.{1,2})}\);/,
+            replace: '{"use strict";$1()=>$2,me:()=>messageEmbed});'
+        }, {
+            match: /function (.{1,2})\((.{1,2})\){var (.{1,2})=.{1,2}\.message,(.{1,2})=.{1,2}\.channel(.{0,300})\(\)\.embedCard(.{0,500})}\)}/,
+            replace: "function $1($2){var $3=$2.message,$4=$2.channel$5().embedCard$6})}\
+var messageEmbed={MessageEmbed:$1};"
+        }]
+    }],
+
+    start() {
+        addAccessory("messageLinkEmbed", props => this.messageEmbedAccessory(props), 4 /* just above rich embeds*/);
+    },
+
+    stop() {
+        // requires a restart to remove the rich embeds because that's not an accessory
+        // and even if they were removed all messages keep their embeds until rerendered
+        // is there a point in having a stop function?
+        removeAccessory("messageLinkEmbed");
+    },
+
+    // the > is kept to be checked for later; cause i have no idea how to make the whole regex fail if it's there
+    messageLinkRegex: /https?:\/\/(?:\w+\.)?discord(?:app)?\.com\/channels\/((?:\d{17,19}|@me)\/\d{17,19}\/\d{17,19}(?:>)?)/,
```

Comment:
```suggestion
    messageLinkRegex: /https?:\/\/(?:\w+\.)?discord(?:app)?\.com\/channels\/((?:\d{17,19}|@me)\/\d{17,19}\/\d{17,19}(?!>))/,
```
use a negative lookahead